src/generate/utils/State_AI.hpp

```diff
@@ -25,17 +26,76 @@ namespace spy::gameplay {
 
             explicit State_AI(const State &s) : State(s) {};
 
+            bool wasHoneyTrapUsed = false; // reset to false for each new operation execution
             double stateChance = 1;
-            std::vector<std::shared_ptr<spy::gameplay::BaseOperation>> operationsLeadingToState;
+            std::vector<std::shared_ptr<BaseOperation>> operationsLeadingToState;
             bool isLeafState = false;
-            // TODO: add evaluation variables here (have to be set by execute)
+            std::vector<spy::gadget::GadgetEnum> collectedGadgets;
+            std::vector<spy::gadget::GadgetEnum> usedGadgets;
+            int chipDiff = 0;
+            std::set<spy::util::UUID> removedClammyClothes;
+            std::set<spy::util::UUID> addedClammyClothes;
+            std::map<spy::util::UUID, int> hpDiff;
+            std::vector<std::optional<double>> observationResult; // if nullopt modify this value (no info about pocket litter)
+            std::optional<double> nuggetResult; // if nullopt modify this value (no info about faction)
+            std::pair<std::optional<double>, std::optional<spy::util::UUID>> chickenfeedResult; // if nullopt modify this value (no info about faction)
+            std::pair<int, std::optional<spy::util::UUID>> mowResult;
+            int unknownGadgetsModifyingSuccess; // for each gadget modify stateChance
+            std::vector<std::pair<std::variant<spy::util::UUID, spy::util::Point>, bool>> removedCocktails;
+            std::vector<std::variant<spy::util::UUID, spy::util::Point>> poisonedCocktails;
+            std::vector<spy::util::Point> destroyedRoulettes;
+            spy::util::Point invertedRoulette;
+            std::vector<spy::util::Point> foggyFields;
+            std::vector<spy::util::Point> destroyedWalls;
+            std::vector<std::pair<spy::util::UUID, std::optional<double>>> spyResult; // if nullopt modify this value (no info about faction)
+            std::vector<spy::util::Point> spyedSafes;
+            spy::util::UUID movedMoledieTo;
```

Comment:
Verstehe ich das richtig, dass hier die änderungen am state geloggt werden sollen?
Es wäre vielleicht sinnvoll, hier nicht `public` von `State` zu erben, um zu verhindern dass man den State verändert ohne diese variablen zu updaten.

Reply:
Dann muss ich die aber auch private machen und extra Methoden schreiben

Reply:
Und ja hier sollen die Änderungen am state geloggt werden

Reply:
Ja genau meine idee wäre, `private` von State zu erben. Wenn das viel aufwand ist und die methoden vom state trotzdem gebraucht werden, lass das so wie es ist.

Reply:
wird nicht in diesem pr gemacht, see #48